nixos/modules/services/continuous-integration/buildkite-agents.nix

```diff
@@ -93,26 +108,26 @@ in
         '';
       };
 
-      openssh =
-        { privateKeyPath = mkOption {
-            type = types.path;
-            description = ''
-              Private agent key.
+      extraSetup = mkOption {
+        type = types.lines;
+        default = "";
+        example = "touch $HOME/test";
+        description = ''
+          Extra commands to while setting up the buildkite dir and config.
+          The directory ownership will be fixed up afterwards.
+        '';
+      };
 
-              A run-time path to the key file, which is supposed to be provisioned
-              outside of Nix store.
-            '';
-          };
-          publicKeyPath = mkOption {
-            type = types.path;
-            description = ''
-              Public agent key.
-
-              A run-time path to the key file, which is supposed to be provisioned
-              outside of Nix store.
-            '';
-          };
-        };
+      sshKeyPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Private agent SSH key.
```

Comment:
The best way to prevent this from going into the store is to use `toString` in an `apply` directly like
```nix
mkOption {
  apply = final: if final == null then null else toString final;
}
```

Then you don't need to worry about using the result of this option without `toString` (and the `toString` below can be removed).